File: backend-go/service/series_service.go
package service

import (
	"context"

	"github.com/sargy/gridpulse/backend-go/model"
	"github.com/sargy/gridpulse/backend-go/repository"
)

type SeriesService struct {
	repo *repository.SeriesRepository
}

func NewSeriesService(repo *repository.SeriesRepository) *SeriesService {
	return &SeriesService{repo: repo}
}

func (s *SeriesService) GetAllSeries(ctx context.Context) ([]model.Series, error) {
	return s.repo.FindAll(ctx)
}

func (s *SeriesService) Count(ctx context.Context) (int64, error) {
	return s.repo.Count(ctx)
}

func (s *SeriesService) SaveAll(ctx context.Context, series []model.Series) error {
	return s.repo.InsertMany(ctx, series)
}

File: backend/src/main/java/com/gridpulse/api/service/SeriesService.java
package com.gridpulse.api.service;

import com.gridpulse.api.model.Series;
import com.gridpulse.api.repository.SeriesRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class SeriesService {

    private final SeriesRepository seriesRepository;

    public SeriesService(SeriesRepository seriesRepository) {
        this.seriesRepository = seriesRepository;
    }

    public List<Series> getAllSeries() {
        return seriesRepository.findAll();
    }

    public long count() {
        return seriesRepository.count();
    }

    public void saveAll(List<Series> seriesList) {
        seriesRepository.saveAll(seriesList);
    }
}
